plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/src/com/google/cloud/tools/eclipse/appengine/standard/java8/FacetChangeListener.java

```diff
@@ -65,10 +65,12 @@ public void handleEvent(IFacetedProjectEvent event) {
       logger.warning(project + ": cannot find appengine-web.xml");
         return;
       }
-    if (project.hasProjectFacet(JavaFacet.VERSION_1_8)) {
-      AppEngineDescriptorTransform.addJava8Runtime(descriptor);
-    } else {
-      AppEngineDescriptorTransform.removeJava8Runtime(descriptor);
+    if (event.getType() == Type.POST_VERSION_CHANGE) {
```

Comment:
Now that you mention it, I do recall there being a facet-discovery mechanism via the `org.eclipse.wst.common.project.facet.core.detectors` extension mechanism.  We should have such a detector that looks for `runtime=java8`.